content/blog/getting-started-with-k8s-part4/index.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "Getting Started with Kubernetes: Stateful Applications"
+date: 2020-07-21
+meta_desc: "How to deploy stateful applications in Kubernetes with Pulumi"
+meta_image: getting-started.png
+authors:
+    - sophia-parafina
+tags:
+    - Kubernetes
+    - StatefulSets
+---
+
+This article is the fourth in a series using infrastructure as code to deploy applications with Kubernetes. This series walks you through:
+
+- [Building a Kubernetes cluster on cloud providers]({{< relref "/blog/getting-started-with-k8s-part1" >}})
+- [Basic application deployment]({{< relref "/blog/getting-started-with-k8s-part2" >}})
+- [Advance application deployment and Helm charts]({{< relref "/blog/getting-started-with-k8s-part3" >}})
+- Stateful applications
+- Networking
+- “Day 2” activities such as migrating node groups.
+
+In the previous post, we examined different methods for deploying applications. We worked through examples of a boilerplate deployment, to one using `ComponentResources` to automate deployment further, and deploying with Helm charts. In this installment, we’ll look at how to deploy stateful applications, such as databases, in Kubernetes. Unlike stateless applications, stateful apps require persistent and durable resources such as storage, which presents scaling and availability challenges.
+
+<!--more-->
+
+## Stateful vs. Stateless
+
+Stateless applications don’t require information from a previous transaction because they are single request and response. For example, the [“Hello World” example](https://www.pulumi.com/blog/getting-started-with-k8s-part2/#examples) that we deployed in an earlier article is a stateless application. The critical thing to remember is that in Kubernetes, any container with that application in any pod can respond to that request.
+Stateful applications can read or store information about previous transactions, and a transaction is executed within the context of a previous extraction. Email is an example of a stateful application, e.g., an email can be unread or read, or it can be grouped according to keywords or by content. Unlike stateless applications in Kubernetes, the requests are sent to the same Pod.
```

Comment:
```suggestion
Stateful applications can read or store information about previous transactions, and a transaction is executed within the context of a previous extraction. Email is an example of a stateful application; an email can be unread or read, or it can be grouped according to keywords or by content. In stateless applications, access to storage must be coordinated, either by sending all requests to the same Pod, or by synchronizing across multiple Pods.
```